jnigen/test/simple_package_test/runtime_test_registrant.dart

```diff
@@ -624,6 +624,47 @@ void registerTests(String groupName, TestRunnerCallback test) {
         expect(MyInterface.$impls, isEmpty);
       });
     }
+    group('Dart exceptions are handled', () {
+      for (final sameThread in [true, false]) {
+        test('on ${sameThread ? 'the same thread' : 'another thread'}',
+            () async {
+          final runnable = MyRunnable.implement(
+            $MyRunnableImpl(
+              run: () {
+                throw UnimplementedError();
```

Comment:
Maybe also throw a String or something, in Dart anything can be thrown, not just Error and Exception subtypes. It would be good to exercise that.